x-pack/plugins/cloud_security_posture/public/pages/configurations/latest_findings/latest_findings_container.tsx

```diff
@@ -107,12 +117,229 @@ const customCellRenderer = (rows: DataTableRecord[]) => ({
   ),
 });
 
+export const FINDINGS_UNIT = (totalCount: number) =>
+  i18n.translate('xpack.csp.findings.unit', {
+    values: { totalCount },
+    defaultMessage: `{totalCount, plural, =1 {finding} other {findings}}`,
+  });
+
+const defaultGroupingOptions = [
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByResource', {
+      defaultMessage: 'Resource',
+    }),
+    key: 'resource.name',
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByRuleName', {
+      defaultMessage: 'Rule name',
+    }),
+    key: 'rule.name',
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByCloudAccount', {
+      defaultMessage: 'Cloud account',
+    }),
+    key: 'cloud.account.name',
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByKubernetesCluster', {
+      defaultMessage: 'Kubernetes cluster',
+    }),
+    key: 'orchestrator.cluster.name',
+  },
+];
+
+const groupingTitle = i18n.translate('xpack.csp.findings.latestFindings.groupBy', {
+  defaultMessage: 'Group findings by',
+});
+
 export const LatestFindingsContainer = ({ dataView }: FindingsBaseProps) => {
+  const getPersistedDefaultQuery = usePersistedQuery(getDefaultQuery);
+  const { urlQuery, setUrlQuery } = useUrlQuery(getPersistedDefaultQuery);
+  const [activePage, setActivePage] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { query } = useBaseEsQuery({
+    dataView,
+    filters: urlQuery.filters,
+    query: urlQuery.query,
+  });
+
+  /**
+   * Reset the active page when the filters or query change
+   * This is needed because the active page is not automatically reset when the filters or query change
+   */
+  useEffect(() => {
+    setActivePage(0);
+  }, [urlQuery.filters, urlQuery.query]);
+
+  const grouping = useGrouping({
+    componentProps: {
+      unit: FINDINGS_UNIT,
+    },
+    defaultGroupingOptions,
+    fields: dataView.fields,
+    groupingId: 'cspLatestFindings',
+    maxGroupingLevels: 1,
+    title: groupingTitle,
+    onGroupChange: () => {
+      setActivePage(0);
+    },
+  });
+
+  const selectedGroup = grouping.selectedGroups[0];
+  const isNoneSelected = isNoneGroup(grouping.selectedGroups);
+
+  const uniqueValue = useMemo(() => `${selectedGroup}-${uuid.v4()}`, [selectedGroup]);
+
+  const groupingQuery = getGroupingQuery({
+    additionalFilters: [query],
+    groupByField: selectedGroup,
+    uniqueValue,
+    from: 'now-1y',
+    to: 'now',
+    pageNumber: activePage * pageSize,
```

Comment:
I'm not sure I understand the `pageNumber` logic here. The group pagination seems to be working, but why when I go to page 5 of the groups, the pageNumber is 50?

Reply:
it's actually `active page index * page size` so page 5 would have a page number of 40, and with the size attribute will query between 40 and 50, I changed the code to be `activePageIndex` instead to clarify that